src/bam/record.rs

```diff
@@ -306,6 +306,19 @@ impl Record {
         }
     }
 
+    // Delete auxiliary tag. Return false if tag doesn't exist
+    pub fn del_aux(&self, tag: &[u8]) -> bool {
```

Comment:
I think it should be `remove_aux` in order to match the signature of HashMap. I also wonder about the return type. It would be more Rusty to return either Option or Result. Option does not make sense here, because we don't return a value. Hence, it should be `Result<(), RemoveAuxError>`, right? `RemoveAuxError` can be defined using the `quick_error` macro as you can see in other places here.